src/main/java/io/vertx/java/spi/cluster/impl/jgroups/JGroupsClusterManager.java

```diff
@@ -0,0 +1,212 @@
+    this.jgroupsConfigurationFile = jgroupsConfigurationFile;
+  }
+
+  @Override
+  public void setVertx(Vertx vertx) {
+    this.vertx = vertx;
+  }
+
+  @Override
+  public <K, V> void getAsyncMultiMap(String name, Handler<AsyncResult<AsyncMultiMap<K, V>>> handler) {
+    logTrace(() -> String.format("Create new AsyncMultiMap [%s] on address [%s]", name, address));
+    checkCluster();
+    cacheManager.createAsyncMultiMap(name, handler);
+  }
+
+  @Override
+  public <K, V> void getAsyncMap(String name, Handler<AsyncResult<AsyncMap<K, V>>> handler) {
+    logTrace(() -> String.format("Create new AsyncMap [%s] on address [%s]", name, address));
+    checkCluster();
+    cacheManager.createAsyncMap(name, handler);
+  }
+
+  @Override
+  public <K, V> Map<K, V> getSyncMap(String name) {
+    logTrace(() -> String.format("Create new SyncMap [%s] on address [%s]", name, address));
+    checkCluster();
+    return cacheManager.createSyncMap(name);
+  }
+
+  @Override
+  public void getLockWithTimeout(String name, long timeout, Handler<AsyncResult<Lock>> handler) {
+    logTrace(() -> String.format("Create new Lock [%s] on address [%s]", name, address));
+    checkCluster();
+    vertx.executeBlocking(
+        future -> {
+          ClusteredLockImpl lock = new ClusteredLockImpl(lockService, name);
+          if (lock.acquire(timeout)) {
+            logDebug(() -> String.format("Lock acquired on [%s]", name));
+            future.complete(lock);
+          } else {
+            future.fail(String.format("Timed out waiting to get lock [%s]", name));
+          }
+        }, handler);
+  }
+
+  @Override
+  public void getCounter(String name, Handler<AsyncResult<Counter>> handler) {
+    logTrace(() -> String.format("Create new counter [%s] on address [%s]", name, address));
+    checkCluster();
+    vertx.executeBlocking(
+        future -> future.complete(new ClusteredCounterImpl(vertx, counterService.getOrCreateCounter(name, 0L))),
+        handler
+    );
+  }
+
+  @Override
+  public String getNodeID() {
+    return address;
+  }
+
+  @Override
+  public List<String> getNodes() {
+    logTrace(() -> String.format("GetNodes on address [%s] with channel view [%s]", address, channel.getViewAsString()));
+    return topologyListener.getNodes();
+  }
+
+  @Override
+  public void nodeListener(NodeListener listener) {
+    logTrace(() -> String.format("Set nodeListener [%s] on address [%s]", listener, address));
+    topologyListener.setNodeListener(listener);
+  }
+
+  @Override
+  public void join(Handler<AsyncResult<Void>> handler) {
+    vertx.executeBlocking((future) -> {
+      synchronized (lock) {
+        if(!active) {
+          try {
+            channel = new JChannel(jgroupsConfigurationFile);
+            topologyListener = new TopologyListener(vertx);
+            channel.setReceiver(topologyListener);
+            channel.connect(CLUSTER_NAME);
+
+            address = channel.getAddressAsString();
```

Comment:
In jgroups, you can set a node name but if you don't do that then what it's called "Address" it's an UUID.